src/infra/core/src/pull_service_impl.rs

```diff
@@ -129,9 +132,14 @@ impl PullServiceImpl {
                     DatasetRefRemote::Alias(alias)
                     | DatasetRefRemote::Handle(DatasetHandleRemote { alias, .. }),
                 ) => DatasetAlias::new(alias.account_name.clone(), alias.dataset_name.clone()),
-                Some(DatasetRefRemote::Url(url)) => {
-                    DatasetAlias::new(None, self.infer_local_name_from_url(url)?)
-                }
+                Some(DatasetRefRemote::Url(url)) => DatasetAlias::new(
+                    if self.dataset_repo.is_multi_tenant() {
```

Comment:
Do we really need to branch here based on the "tenancy"? I thought the single-tenant repo will function by using certain hard-coded account name (e.g. `kamu`) for the current user. So that all service code can always work in a multi-tenant mode but this account name will simply be ignored when forming a dataset fs path.

It would be nice to have all services just assume we are always multi-tenant and not make any special cases.

Reply:
This is required when 2 users have datasets with identical names, and we are pulling via alternative user (`-a` switch) without specifying account name in the reference:

  `kamu -a wasya pull alberta.case-details`